KBI0024: Create and update a 7-Zip archive in a RIA store (#90)
psychoinformatics-de/knowledge-base: kbi/0024/index.rst

.. index::
   single: RIA; archive
.. highlight:: console

KBI0024: Create and update a 7-Zip archive in a RIA store
=========================================================

:authors: Laura Waite <laura@waite.eu>, Michael Hanke <michael.hanke@gmail.com>
:discussion: <link>
:keywords: RIA, archive, 7-Zip, inodes
:software-versions: datalad_0.18.3

When working with `RIA stores`_, it is possible to compress the
``annex/objects`` into `7-zip`_ archives. With this approach, the entire
annex object store can be put into an archive and remain fully accessible while

> minimizing inodes, regardless of file number and size. This is beneficial

Reviewer: ```suggestion
minimizing `inodes`_, regardless of file number and size. This is beneficial
```

Maybe also add a link to the term definition - to me this used to be a foreign sounding word. Not necessary though, we use many other technical terms across the KBI, and it's a quick search away.

If this suggestion is accepted, a link should go at the bottom of the page.

Author: Good idea.

for compression gains or when operating on HPC-systems with inode limitations.

This document describes how to create a 7-Zip archive for an existing RIA
store using the script shown below. The script does the following:

> 1. Remove unnecessary files and directories within the RIA store (e.g. hooks,
>    etc). This aspect has nothing to do with creating the archive, but is useful
>    for reducing the number of inodes.

Reviewer: ```suggestion
1. Remove non-essential files and directories within the RIA store (e.g. hooks,
   etc). This aspect has nothing to do with creating the archive, but is useful
   for reducing the number of inodes.
```

To me _unnecessary_ begs the question why they are there in the first place, and non-essential sounds better. But maybe it's just me - in any case, this suggestion is non-essential :)

  │       │   ├── info
  │       │   └── pack
  │       ├── refs
  │       │   ├── heads
  │       │   │   ├── git-annex
  │       │   │   └── master
  │       │   └── tags
  │       └── ria-layout-version
  ├── error_logs
  └── ria-layout-version

Let's verify that the archive was updated successfully with the new content,
using the dataset we started with::

  $ cd my_dataset

This dataset only has one file (``file1.txt``)::

  $ tree
  .
  └── file1.txt -> .git/annex/objects/v2/2W/MD5E-s15--af1cdf0b10caa12cf13312f7bb4215df.txt/MD5E-s15--af1cdf0b10caa12cf13312f7bb4215df.txt

Run ``datalad update`` to bring in the updates from the RIA store (i.e. ``file2.txt``)::

  $ datalad update --merge
  $ tree
  .
  ├── file1.txt -> .git/annex/objects/v2/2W/MD5E-s15--af1cdf0b10caa12cf13312f7bb4215df.txt/MD5E-s15--af1cdf0b10caa12cf13312f7bb4215df.txt
  └── file2.txt -> .git/annex/objects/Pf/vq/MD5E-s15--7a593f3460f1efc629489d5a9e86c7b0.txt/MD5E-s15--7a593f3460f1efc629489d5a9e86c7b0.txt

.. literalinclude:: cleanup.sh
   :language: bash
   :linenos:
   :caption: cleanup.sh

.. _ria stores: http://handbook.datalad.org/en/latest/beyond_basics/101-147-riastores.html

> .. _7-Zip: https://www.7-zip.org/

Reviewer: ```suggestion
.. _7-Zip: https://www.7-zip.org/
.. _inodes: https://en.wikipedia.org/wiki/Inode
```

Related to my first suggestion. I'm fine with rejecting or using a different target.

.. index::
   single: RIA; archive
.. highlight:: console

KBI0024: Create and update a 7-Zip archive in a RIA store
=========================================================

:authors: Laura Waite <laura@waite.eu>, Michael Hanke <michael.hanke@gmail.com>

> :discussion: <link>

Reviewer 2: ```suggestion
:discussion: https://github.com/psychoinformatics-de/knowledge-base/issues/47
```
